OptimizelySDKCore/OptimizelySDKCore/OPTLYEventBuilder.m

```diff
@@ -220,185 +223,144 @@ - (NSNumber *)numericValue:(OPTLYProjectConfig *)config value:(NSObject *)value
             // JSON booleans "true" and "false" instead of JSON numbers.
             // These aren't integers, so shouldn't be sent.
             answer = nil;
-            [config.logger logMessage:OPTLYLoggerMessagesNumericValueInvalidBoolean withLevel:OptimizelyLogLevelWarning];
+            [self.config.logger logMessage:OPTLYLoggerMessagesNumericValueInvalidBoolean withLevel:OptimizelyLogLevelWarning];
         } else {
             // Require real numbers (not infinite or NaN).
             double doubleValue = [(NSNumber*)value doubleValue];
             if (isfinite(doubleValue)) {
                 answer = (NSNumber*)value;
             } else {
                 answer = nil;
-                [config.logger logMessage:[NSString stringWithFormat:OPTLYLoggerMessagesNumericValueInvalidFloat, value] withLevel:OptimizelyLogLevelWarning];
+                [self.config.logger logMessage:[NSString stringWithFormat:OPTLYLoggerMessagesNumericValueInvalidFloat, value] withLevel:OptimizelyLogLevelWarning];
             }
         }
     } else if ([value isKindOfClass:[NSString class]]) {
         // cast strings to double
         double doubleValue = [(NSString*)value doubleValue];
         if (isfinite(doubleValue)) {
-            [config.logger logMessage:[NSString stringWithFormat:OPTLYLoggerMessagesNumericValueString, value] withLevel:OptimizelyLogLevelWarning];
+            [self.config.logger logMessage:[NSString stringWithFormat:OPTLYLoggerMessagesNumericValueString, value] withLevel:OptimizelyLogLevelWarning];
             answer = [NSNumber numberWithDouble:doubleValue];
         } else {
-            [config.logger logMessage:[NSString stringWithFormat:OPTLYLoggerMessagesNumericValueInvalidString, value] withLevel:OptimizelyLogLevelWarning];
+            [self.config.logger logMessage:[NSString stringWithFormat:OPTLYLoggerMessagesNumericValueInvalidString, value] withLevel:OptimizelyLogLevelWarning];
         }
     } else {
         // all other objects can't be cast to double
-        [config.logger logMessage:OPTLYLoggerMessagesNumericValueInvalid withLevel:OptimizelyLogLevelWarning];
+        [self.config.logger logMessage:OPTLYLoggerMessagesNumericValueInvalid withLevel:OptimizelyLogLevelWarning];
     };
     return answer;
 }
 
-- (NSDictionary *)createCommonParams:(OPTLYProjectConfig *)config userId:(NSString *)userId
-                          attributes:(NSDictionary<NSString *, NSString *> *)attributes {
-    NSMutableDictionary *params = [NSMutableDictionary new];
+- (NSDictionary *)createCommonParamsForUser:(NSString *)userId
```

Comment:
not reviewing for now, will check it via fullstack-suite. Assuming you copied from the PR to correct EventBuilder payload.

Reply:
Just removed ProjectConfig from the method signature.

Reply:
Please also do some manual E2E on this PR and ensure that events are getting converted correctly and show up on the results page. That gives us much better guarantee over the compat-suite. Only asking for this because refactoring event builder is always risky :)

Reply:
Sure Mike, I will take care of it.